SudokuSolver/SudokuSolver/SudokuSolver.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Extensions;
+
+namespace SudokuSolver
+{
+    public class SudokuSolver
+    {
+        public class EmptySudokuCube
+        {
+            public int Row { get; init; }
+            public int Col { get; init; }
+
+            public List<int> PossibleValues { get; init; }
+            public EmptySudokuCube(int row, int col)
+            {
+                Row = row;
+                Col = col;
+                PossibleValues = new List<int>();
+            }
+        }
+
+        private int[,] _grid;
+        private int _maxRows;
+        private int _maxCols;
+        private int _rowSqrt;
+        private int _colSqrt;
+
+        private List<EmptySudokuCube> _emptyCubes;
+
+        private void InitEmptyCubes()
+        {
+            _emptyCubes = new List<EmptySudokuCube>();
+
+            for (int i = 0; i < _maxRows; i++)
+            {
+                for (int j = 0; j < _maxCols; j++)
+                {
+                    if (_grid[i, j] == 0)
+                    {
+                        var eSC = new EmptySudokuCube(i, j);
+                        // Check all legal nums and add them to cube's possible values
+                        eSC.PossibleValues.AddRange(
```

Comment:
skoro zawsze to EmptySudokuCube ma mieć w sobie listę possible values to lepiej wymagać tej listy w konstruktorze od razu. Zmniejsza to szansę na to, że ktoś się pomyli i zapomni tej listy zainicjalizować po stworzeniu tego obiektu. Jak będzie wymagana jako argument konstruktora to kompilator będzie pilnował.